
# HG changeset patch
# User Tomas Mraz <tmraz@redhat.com>
# Date 1348152857 -7200
# Node ID dc06ede8e857bf0a623121547f93154c03984bf6
# Parent  98d012ea527ab04c7a0c37169f0ed323118d2345
Use the new smb.conf idmap config range syntax (#850824)

diff -r 98d012ea527a -r dc06ede8e857 authconfig.py
--- a/authconfig.py	Tue Sep 18 20:45:54 2012 +0200
+++ b/authconfig.py	Thu Sep 20 16:54:17 2012 +0200
@@ -224,10 +224,8 @@
 			help=_("names of servers to authenticate against"))
 		parser.add_option("--smbworkgroup", metavar=_("<workgroup>"),
 			help=_("workgroup authentication servers are in"))
-		parser.add_option("--smbidmapuid", metavar=_("<lowest-highest>"),
+		parser.add_option("--smbidmaprange", "--smbidmapuid", "--smbidmapgid", metavar=_("<lowest-highest>"),
 			help=_("uid range winbind will assign to domain or ads users"))
-		parser.add_option("--smbidmapgid", metavar=_("<lowest-highest>"),
-			help=_("gid range winbind will assign to domain or ads users"))
 		parser.add_option("--winbindseparator", metavar="<\\>",
 			help=_("the character which will be used to separate the domain and user part of winbind-created user names if winbindusedefaultdomain is not enabled"))
 		parser.add_option("--winbindtemplatehomedir", metavar="</home/%D/%U>",
@@ -481,8 +479,7 @@
 			"smbservers":"smbServers",
 			"smbsecurity":"smbSecurity",
 			"smbrealm" : "smbRealm",
-			"smbidmapuid":"smbIdmapUid",
-			"smbidmapgid":"smbIdmapGid",
+			"smbidmaprange":"smbIdmapRange",
 			"winbindseparator":"winbindSeparator",
 			"winbindtemplatehomedir":"winbindTemplateHomedir",
 			"winbindtemplateprimarygroup":"winbindTemplatePrimaryGroup",
diff -r 98d012ea527a -r dc06ede8e857 authinfo.py
--- a/authinfo.py	Tue Sep 18 20:45:54 2012 +0200
+++ b/authinfo.py	Thu Sep 20 16:54:17 2012 +0200
@@ -1291,8 +1291,7 @@
 		self.smbRealm = ""
 		self.smbServers = ""
 		self.smbSecurity = ""
-		self.smbIdmapUid = ""
-		self.smbIdmapGid = ""
+		self.smbIdmapRange = ""
 
 		self.winbindSeparator = ""
 		self.winbindTemplateHomedir = ""
@@ -1421,8 +1420,8 @@
 		("enableLDAPAuth", "b"), ("enableIPAv2", "b")]),
 	SaveGroup(self.writeSmartcard, [("smartcardAction", "i"), ("smartcardModule", "c")]),
 	SaveGroup(self.writeWinbind, [("smbWorkgroup", "i"), ("smbServers", "i"),
-		("smbRealm", "c"), ("smbSecurity", "i"), ("smbIdmapUid", "i"),
-		("smbIdmapGid", "i"), ("winbindSeparator", "c"), ("winbindTemplateHomedir", "c"),
+		("smbRealm", "c"), ("smbSecurity", "i"), ("smbIdmapRange", "i"),
+		("winbindSeparator", "c"), ("winbindTemplateHomedir", "c"),
 		("winbindTemplatePrimaryGroup", "c"), ("winbindTemplateShell", "c"),
 		("winbindUseDefaultDomain", "b"), ("winbindOffline", "b")]),
 	SaveGroup(self.writeNSS, [("enableDB", "b"), ("enableDirectories", "b"), ("enableWinbind", "b"),
@@ -1943,19 +1942,12 @@
 			self.setParam("smbSecurity", tmp, ref)
 		if not self.smbSecurity:
 			self.smbSecurity = "user"
-		tmp = self.readWinbindGlobal("idmap uid")
+		tmp = self.readWinbindGlobal("idmap config * : range")
 		if tmp:
-			self.setParam("smbIdmapUid", tmp, ref)
-		if not self.smbIdmapUid:
+			self.setParam("smbIdmapRange", tmp, ref)
+		if not self.smbIdmapRange:
 			# 2^24 to 2^25 - 1 should be safe
-			self.smbIdmapUid = "16777216-33554431"
-		tmp = self.readWinbindGlobal("idmap gid")
-		if tmp:
-			self.setParam("smbIdmapGid", tmp, ref)
-
-		if not self.smbIdmapGid:
-			# 2^24 to 2^25 - 1 should be safe
-			self.smbIdmapGid = "16777216-33554431"
+			self.smbIdmapRange = "16777216-33554431"
 		tmp = self.readWinbindGlobal("winbind separator")
 		if tmp:
 			self.setParam("winbindSeparator", tmp, ref)
@@ -3304,16 +3296,11 @@
 			output += self.smbSecurity
 			output += "\n"
 			wrotesecurity = True
-		if self.smbIdmapUid:
-			output += "   idmap uid = "
-			output += self.smbIdmapUid
+		if self.smbIdmapRange:
+			output += "   idmap config * : range = "
+			output += self.smbIdmapRange
 			output += "\n"
-			wroteidmapuid = True
-		if self.smbIdmapGid:
-			output += "   idmap gid = "
-			output += self.smbIdmapGid
-			output += "\n"
-			wroteidmapgid = True
+			wroteidmaprange = True
 		if self.winbindSeparator:
 			output += "   winbind separator = "
 			output += self.winbindSeparator
@@ -4040,8 +4027,7 @@
 		print " SMB security = \"%s\"" % self.smbSecurity
 		print " SMB realm = \"%s\"" % self.smbRealm
 		print " Winbind template shell = \"%s\"" % self.winbindTemplateShell
-		print " SMB idmap uid = \"%s\"" % self.smbIdmapUid
-		print " SMB idmap gid = \"%s\"" % self.smbIdmapGid
+		print " SMB idmap range = \"%s\"" % self.smbIdmapRange
 		print "nss_sss is %s by default" % formatBool(self.enableSSSD)
 		print "nss_wins is %s" % formatBool(self.enableWINS)
 		print "nss_mdns4_minimal is %s" % formatBool(self.enableMDNS)

